### PR TITLE
Allow case insensitive paths for determining `PSModulePath`

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

When comparing paths on Windows for determining the PSModulePath, the comparison should be case insensitive.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ internal static string GetWindowsPowerShellModulePath()
             }
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
TestCustomPipeNameCreation
```powershell
Assert.True() Failure
Expected: True
Actual:   False
   at PSTests.Parallel.NamedPipeTests.TestCustomPipeNameCreation() in D:\a\1\s\test\xUnit\csharp\test_NamedPipe.cs:line 25
```

Comment (PoshChan) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-macos`
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.8.92', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

**Review by SteveL-MSFT — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: @adityapatwardhan, your last commit had 2 failures in `PowerShell-CI-macos`
Test-Connection.Ping.Force IPv4 with implicit PingOptions
```powershell
Expected exactly '127.0.0.1', but got 10.79.0.23.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 83
83:             $result[0].Address | Should -BeExactly $realAddress
```
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '127.0.0.1', but got 10.79.0.23.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs [hunk 2]
```diff
@@ -1250,7 +1250,7 @@ internal static string GetWindowsPowerShellModulePath()
...
 
             // PowerShell specific paths including if set in powershell.config.json file we want to exclude
-            var excludeModulePaths = new HashSet<string> {
+            var excludeModulePaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase) {
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: Should we document the behavior n Unix?

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs (hunk 2 above)
Comment (adityapatwardhan) [reply src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: The code is within `#if !UNIX` block

**Comment by adityapatwardhan** [comment]
@PoshChan please retry macos

**Comment by PoshChan** [comment]
@adityapatwardhan, successfully started retry of `PowerShell-CI-macOS`

File: src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs]: @adityapatwardhan, your last commit had 2 failures in `PowerShell-CI-macos`
Test-Connection.Ping.Force IPv4 with implicit PingOptions
```powershell
Expected exactly '127.0.0.1', but got 10.79.0.23.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 83
83:             $result[0].Address | Should -BeExactly $realAddress
```
Test-Connection.TraceRoute.TraceRoute works
```powershell
Expected exactly '10.79.0.59', but got 127.0.0.1.
at <ScriptBlock>, /Users/runner/runners/2.165.2/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1: line 257
257:             $result[0].TargetAddress | Should -BeExactly $realAddress
```

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in one hour

**Comment by PoshChan** [comment]
@TravisEz13, I do not understand: Please remind me in one hour

<details>
<summary>Commands available in this repo for you:
</summary>
<ul><li>
<code>retry &lt;target&gt;</code> this will attempt to retry only the failed jobs for the target pipeline, <code>restart</code> can be used in place of <code>retry</code>
</li>
<li>
<code>rebuild &lt;target&gt;</code> this will perform a complete rebuild of the target pipeline, <code>rerun</code> can be used in place of <code>rebuild</code> Supported values for &lt;target&gt; which can be a comma separated list are: <code>static</code>,<code>windows</code>,<code>macos</code>,<code>linux</code>,<code>all</code>
</li>
<li>
<code>get failures</code> this will attempt to get the latest failures for all of the target pipelines
</li>
<li>
<code>remind me in &lt;value&gt; &lt;units&gt;</code> this will create a reminder that will be posted after the specified duration &lt;value&gt; is a number, and &lt;units&gt; can be <code>minutes</code>, <code>hours</code>, or <code>days</code> (singular or plural)
</li>

</ul>
</details>

**Pull request merged**